web/src/user_group_edit.ts

```diff
@@ -626,47 +634,293 @@ function populate_data_for_removing_user_group_permissions(
     return data;
 }
 
-export function update_setting_in_group_permissions_panel(
-    $setting_elem: JQuery,
+export function add_assigned_permission_to_permissions_panel(
+    setting_name: string,
+    $subsection_elem: JQuery,
+    subsection_settings: string[],
+    rendered_checkbox_html: string,
+): void {
+    const $setting_elem = $subsection_elem.find(`.prop-element[name="${CSS.escape(setting_name)}"`);
+    if ($setting_elem.length > 0) {
+        // If there is already a checkbox for that permission, there can
+        // still be changes in permission whether the permission can be
+        // removed from the panel or not. So, just replace it with the
+        // newly rendered checkbox.
+        $setting_elem.closest(".input-group").replaceWith($(rendered_checkbox_html));
+        return;
+    }
+
+    if ($subsection_elem.hasClass("hide")) {
+        $subsection_elem.removeClass("hide");
+    }
+
+    if ($subsection_elem.closest(".group-permissions-section").hasClass("hide")) {
+        $subsection_elem.closest(".group-permissions-section").removeClass("hide");
+    }
+
+    if (!$(".group-assigned-permissions .no-permissions-for-group-text").hasClass("hide")) {
+        $(".group-assigned-permissions .no-permissions-for-group-text").addClass("hide");
+    }
+
+    let insert_position = 0;
+    for (const name of subsection_settings) {
+        if (name === setting_name) {
+            break;
+        }
+        if ($subsection_elem.find(`.prop-element[name="${CSS.escape(name)}"`).length > 0) {
+            insert_position = insert_position + 1;
+        }
+    }
+
+    if (insert_position === 0) {
+        $subsection_elem.find(".subsection-settings").prepend($(rendered_checkbox_html));
```

Comment:
All calls to `prepend`, `append`, and similar functions are using rendered handlerbar templates and that ensures that escaping is done correctly.

Reply:
Yeah, this is great.